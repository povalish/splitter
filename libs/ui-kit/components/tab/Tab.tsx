import { cx } from 'class-variance-authority';
import { IWebTab } from './Tab.interface';
import { tabClasses, textClasses } from './Tab.classes';

//
//

export const Tab: React.FC<IWebTab> = ({ label, tabId, panelId, active = false, onSelect }) => (
  <button
    type='button'
    onClick={onSelect}
    className={cx(textClasses({ active }), tabClasses({ active }))}
    id={tabId}
    role='tab'
    aria-selected={active ? 'true' : 'false'}
    aria-controls={panelId}
  >
    {label}
  </button>
);
