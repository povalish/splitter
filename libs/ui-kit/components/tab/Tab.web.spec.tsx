import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Tab } from './Tab';

//
//

describe('Tab component', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Tab label='focusing' tabId='tab-1' panelId='panel-1' active />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call onSelect', () => {
    const handleSelect = jest.fn();
    render(<Tab label='focusing' tabId='tab-1' panelId='panel-1' active onSelect={handleSelect} />);

    fireEvent.click(screen.getByText(/focusing/i));
    expect(handleSelect).toHaveBeenCalledTimes(1);
  });
});
