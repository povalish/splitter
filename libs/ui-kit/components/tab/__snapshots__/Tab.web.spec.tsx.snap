// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component should render successfully 1`] = `
<body>
  <div>
    <button
      aria-controls="panel-1"
      aria-selected="true"
      class="
  text-[20px] text-colorText font-bold whitespace-nowrap
  md:text-[24px]
 
  py-2 px-3
  transition-all
  active:scale-95
  hover:scale-105
  focus:text-colorAccent focus:opacity-100 focus:text-shadow focus:outline-none
 opacity-100"
      id="tab-1"
      role="tab"
      type="button"
    >
      focusing
    </button>
  </div>
</body>
`;
